src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -145,302 +139,203 @@ CHIP_ERROR WiFiManager::Init()
         return CHIP_NO_ERROR;
     });
 
-    ChipLogDetail(DeviceLayer, "wpa_supplicant has been initialized");
+    net_mgmt_init_event_callback(&mWiFiMgmtClbk, WifiMgmtEventHandler, sWifiManagementEvents);
+    net_mgmt_add_event_callback(&mWiFiMgmtClbk);
+
+    ChipLogDetail(DeviceLayer, "WiFiManager has been initialized");
 
     return CHIP_NO_ERROR;
 }
-
-CHIP_ERROR WiFiManager::AddNetwork(const ByteSpan & ssid, const ByteSpan & credentials)
+CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanCallback callback)
```

Comment:
Looks like the ScanCallback is not registered and the scan response is not translated to Matter data types. Therefore the scan command sent from the controller will not work. I will try to fix that.